hypothesis-python/src/hypothesis/internal/reflection.py

```diff
@@ -166,27 +166,15 @@ def get_signature(
             return sig.replace(
                 parameters=[v for k, v in sig.parameters.items() if k != "self"]
             )
-    sig = _inspect_signature(target, follow_wrapped=follow_wrapped, eval_str=eval_str)
-    check_signature(sig)
-    return sig
-
-
-# eval_str is only supported by Python 3.10 and newer
-if sys.version_info[:2] >= (3, 10):
-
-    def _inspect_signature(
-        target: Any, *, follow_wrapped: bool = True, eval_str: bool = False
-    ) -> inspect.Signature:
-        return inspect.signature(
+    # eval_str is only supported by Python 3.10 and newer
+    if sys.version_info[:2] >= (3, 10):
+        sig = inspect.signature(
             target, follow_wrapped=follow_wrapped, eval_str=eval_str
         )
-
-else:
-
-    def _inspect_signature(
-        target: Any, *, follow_wrapped: bool = True, eval_str: bool = False
-    ) -> inspect.Signature:  # pragma: no cover
-        return inspect.signature(target, follow_wrapped=follow_wrapped)
+    else:
+        sig = inspect.signature(target, follow_wrapped=follow_wrapped)
```

Comment:
Nice solution 🙂 

Just needs a `# pragma: no cover` comment on one of these lines - we configure coverage to ignore branching on `sys.version`, but that doesn't cover the `else:`.